### DISCUSSION TITLE
Called workflows cannot be queued onto self-hosted runners across organisations/enterprises. Failed to queue this job. Labels: 'ubuntu-latest'

### DISCUSSION THREAD

**Discussion by briantist** [community/community#25214] [Enterprise]
<p>Reusable workflows being called from another repository are failing with the error in the title.</p>
<p>This is the same problem as: <a href="https://github.community/t/ubuntu-latest-runner-not-found-when-using-workflow-call-between-repositories/204957/16" class="inline-onebox">'ubuntu-latest' runner not found when using workflow_call between repositories - #16 by mohamed3114</a><br>
but that has already been marked solved, and this is a new issue or some kind of regression, because it only just started happening.</p>
<p>There seems to be something to <a href="https://github.community/t/ubuntu-latest-runner-not-found-when-using-workflow-call-between-repositories/204957/5">this reply</a>, which mentions that this only happens when the concurrency limit is reached. When I retry the jobs with available capacity, they work, but every time there’s contention, these jobs fail now.</p>
<p>It’s very disruptive and means I need to babysit my jobs so that I can re-submit the ones that use reusable workflows from another repository, once I can see that they won’t be queued. And in the meantime, I am getting emailed about these failed jobs too.</p>
<p>I should add, there are NO self-hosted runners in use in my repository.</p>
<p>cc <a class="mention" href="/jenschelkopf">@jenschelkopf</a> (since you resolved the previous issue)</p>

**Comment by briantist** [comment]
<p>Can anyone help with this? It’s so disruptive, every PR, commit, merge has these jobs that originate from reusable workflows failing and forcing me to watch and wait until there’s capacity before I re-run them…</p>

**Comment by erzz** [comment]
<p>Well this is my “me-too” post.  You are absolutely right - it seems when a large number (~15-25 concurrent jobs) for a specific repo, this pops up.</p>
<p>You are absolutely right that is very disruptive that my production pipeline fails because 2 people just pushed to feature branches.</p>
<p>So much for velocity!</p>

**Comment by samccann** [comment]
<p>Don’t have much to add other than this is holding up a nifty new feature for our repos as well.</p>

**Comment by acozine** [comment]
<p>Adding my voice to the chorus here. Any idea when this bug will be addressed?</p>

**Comment by ethomson** [comment]
<p>Thanks for reporting this; the team is investigating.  I hope to have more to share soon.</p>

**Comment by briantist** [comment]
<p>Great to hear! Looking forward to a fix… :pray:</p>

**Comment by SimonDarksideJ** [comment]
<p>3 days later and still no response.<br>
Finally got our workflows up and “accepted” by the GitHub Actions for our pipeline and getting the same error.</p>
<p>We have a Self-Hosted agent running with the tags required but GitHub actions is refusing to pick up and run it on the runner.</p>
<p>This exact flow was working perfectly fine on a personal / private repo with exactly the same agent and machine.  Only thing that has changed is that it is now an Organisation runner.</p>
<p>Ok, re-read the original report and it notes that they are not using self-hosted runners, my bad.  But issue persists</p>

**Comment by briantist** [comment]
<p>Yeah based on the error message it seems known/intentional that cross-org on self-hosted runners is not supported (though I have no idea why, that doesn’t seem intuitive at all); however this thread/topic is mainly to report that this error is newly, erroneously returned on GitHub runners. I recommend finding or starting a new topic to request a change in the functionality for cross-org self-hosted runners.</p>

**Comment by SimonDarksideJ** [comment]
<p>I have here :smiley:</p>
<p><a href="https://github.community/t/called-workflows-cannot-be-queued-onto-self-hosted-runners-across-organisations-enterprises-failed-to-queue-this-job-labels-windows/232181/2">Called workflows cannot be queued onto self-hosted runners across organisations/enterprises. Failed to queue this job. Labels: ‘windows’ - Code to Cloud / GitHub Actions - GitHub Community</a></p>

**Comment by briantist** [comment]
<p><a class="mention" href="/ethomson">@ethomson</a> any update?<br>
Does this warrant an issue being created in the runner repository?</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/actions/runner/issues">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/actions/runner/issues" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/1260690/181089619-20e01184-7719-44fd-9b71-9378d7c3091a.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/5/05b9e662ea201ddfe53519edada682d74ccc6e74_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/5/05b9e662ea201ddfe53519edada682d74ccc6e74_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/0/5/05b9e662ea201ddfe53519edada682d74ccc6e74.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/5/05b9e662ea201ddfe53519edada682d74ccc6e74_2_10x10.png"></div>

<h3><a href="https://github.com/actions/runner/issues" target="_blank" rel="noopener nofollow ugc">Issues · actions/runner</a></h3>

  <p>The Runner for GitHub Actions :rocket:. Contribute to actions/runner development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Or is there any other way I can track progress on it?</p>

**Comment by briantist** [comment]
<p><a class="mention" href="/ethomson">@ethomson</a> please can someone help? The problem is not only still occurring, it’s getting way worse. The jobs are now failing even more quickly, even when there are no other workflows running. They sit for a little over 1 minute and then die with the false self-hosted runner message, and all I can do is to keep trying over and over again and hope it runs.<br>
This run, which has 3 jobs that run <em>successively</em> not in parallel, with nothing else running at all, <strong>took 11 attempts before I could finally run it!</strong></p><aside class="onebox githubactions" data-onebox-src="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1986738843">
  <header class="source">

      <a href="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1986738843" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
      <svg width="24" height="24" class="github-icon github-icon-success" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm16.28-2.72a.75.75 0 00-1.06-1.06l-5.97 5.97-2.47-2.47a.75.75 0 00-1.06 1.06l3 3a.75.75 0 001.06 0l6.5-6.5z"></path></svg>



  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1986738843" target="_blank" rel="noopener nofollow ugc">empty</a>
    </h4>

      <div class="github-info">
        Collection Docs <span class="github-run-number">#811</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>This experience is really awful. It’s killing my velocity and motivation to work on anything. Please help!!</p>

**Comment by briantist** [comment]
<p><a class="mention" href="/ethomson">@ethomson</a> please, is there any update?</p>

**Comment by ethomson** [comment]
<p>Hi Brian - I don’t have an ETA but I checked in with the engineering team and they believe they have identified the problem and occasions on which this occurs. They should be working on a fix now. Apologies for the delay.</p>

**Comment by briantist** [comment]
<p>Thanks <a class="mention" href="/ethomson">@ethomson</a> , did something change yesterday/today? I’m seeing one of these jobs properly queue without failing, so that’s hopeful… :pray:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/ansible-collections/community.hashi_vault/runs/5776610083">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/ansible-collections/community.hashi_vault/runs/5776610083" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/1260690/181089622-dda58026-e66d-4b7b-a6c5-46f6e67c3e18.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/ansible-collections/community.hashi_vault/runs/5776610083" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 73 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by ethomson** [comment, marked as answer]
<p>Yes, the fixes for this should be completely rolled out to all customers now. Thanks for letting us know that it’s working and thanks again for your patience.</p>

**Reply by artakvg** [reply]
@ethomson I am still seeing this issue.
I am using [this](https://github.com/dooqod/workflow-actions/blob/main/.github/workflows/docker_build_push_nodeservice.yml) public reusable workflow from my private repo with a self hosted runner configured (I can see the status IDLE), but when running workflow from my private repo , its wailing immediately with the error in the screenshot
 
<img width="1326" alt="Screenshot 2022-12-18 at 03 52 58" src="https://user-images.githubusercontent.com/2362491/208270477-159c3590-78ee-46d5-98de-c5082a5a76a7.png">

One more thing:
I am using the reusable workflow with the input param like in the pic below 
<img width="753" alt="Screenshot 2022-12-18 at 03 57 30" src="https://user-images.githubusercontent.com/2362491/208270627-0fc4de79-2165-49d2-9d8a-13d6ea48e36d.png">

**Reply by cyrixmorten** [reply]
Experiencing the same on private repo using shared workflows from https://github.com/nrwl/ci and large runners provisioned by Github. 

```
name: Build
on:
  push:
    branches:
      - main
      - develop

  pull_request:
    types: [opened, synchronize, reopened]

  # Allows to run this workflow manually from the Actions tab
  workflow_dispatch:

concurrency:
  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
  cancel-in-progress: true

jobs:

  agents:
    name: Agents
    uses: nrwl/ci/.github/workflows/nx-cloud-agents.yml@v0.11.2
    secrets:
      NX_CLOUD_AUTH_TOKEN: ${{ secrets.NX_CLOUD_AUTH_TOKEN }}
    with:
      runs-on: ubuntu-latest-16-cores
      number-of-agents: 3
      working-directory: 'workspaces'
      environment-variables: |
        TZ=Europe/Copenhagen


  main:
    name: Main
    uses: nrwl/ci/.github/workflows/nx-cloud-main.yml@v0.11.2
    secrets:
      NX_CLOUD_AUTH_TOKEN: ${{ secrets.NX_CLOUD_AUTH_TOKEN }}
    with:
      runs-on: ubuntu-latest-4-cores
      number-of-agents: 3
      working-directory: 'workspaces'
      environment-variables: |
        TZ=Europe/Copenhagen
      parallel-commands-on-agents: |
        pnpm nx affected --target=test --ci --parallel 16
        pnpm nx affected --target=build-types --parallel 16
        pnpm nx affected --target=build-lib --parallel 16
```

![image](https://user-images.githubusercontent.com/6650168/213134522-70cc0795-c3c7-4f8c-b6e6-514bae02995b.png)

**Reply by th-le** [reply]
hi @ethomson, which version that has the fix you mentioned ? I'm using github enterprise 3.5 with self hosted runners and still see this issue

**Reply by Michsior14** [reply]
I have the same issue as @cyrixmorten. Is there already any solution for this?

**Reply by jazzlyn** [reply]
ethomson is not part of github anymore and this discussion got "answered". so I assume there will be no further attention on this. anyways not working for me neither, same issue as all of you above. it seems not possible to use a public reusable workflow with a self-hosted runner.

**Comment by Saidfatah** [comment]
<p>Hi ethomson , seems like this is still occurring I’ve been facing it all day , here is my .yml file :</p>
<pre><code class="lang-auto">name: Super-Linter

on: push

jobs:
  super-lint:
    name: Lint code base
    runs-on: self-hosted
    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      - name: Run Super-Linter
        uses: github/super-linter@v3
        env:
          DEFAULT_BRANCH: main
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>

**Comment by briantist** [comment]
<p>This looks it might be a different issue. This thread is specifically about reusable workflows, your workflow is not a definition of a resuable workflow, and it is not consuming one either. You may be better served posting a new thread for this one.</p>
